Submitted By:            Randy McMurchy <randy_at_linuxfromscratch_dot_org>
Date:                    2005-08-29
Initial Package Version: 2.0
Upstream Status:         Unknown
Origin:                  http://dev.gentoo.org/~halcy0n/overlay/media-libs/faad2/files/
Description:             Fixes several build issues
                         1) Incorrect space in the Makefile
                         2) Adds checks for recent mpeg4ip installation
                         3) XMMS plugin access to internal mpeg4ip
                         4) corrects problems with all version of GCC up to
                            and including GCC-4.0.1


diff -Naur faad2-orig/Makefile.am faad2/Makefile.am
--- faad2-orig/Makefile.am	2004-02-06 16:04:39.000000000 +0000
+++ faad2/Makefile.am	2005-08-29 15:33:11.000000000 +0000
@@ -1,8 +1,3 @@
 SUBDIRS = libfaad common frontend plugins
 
 EXTRA_DIST = faad2.spec
-
-rpm:    Makefile
-       make dist
-       $(RPMBUILD) -ta $(PACKAGE)-$(VERSION).tar.gz
-       rm $(PACKAGE)-$(VERSION).tar.gz

diff -Naur faad2-orig/common/mp4ff/mp4ffint.h faad2/common/mp4ff/mp4ffint.h
--- faad2-orig/common/mp4ff/mp4ffint.h	2004-01-14 20:50:22.000000000 +0000
+++ faad2/common/mp4ff/mp4ffint.h	2005-08-29 15:33:41.000000000 +0000
@@ -301,7 +301,7 @@
 mp4ff_t *mp4ff_open_edit(mp4ff_callback_t *f);
 #endif
 void mp4ff_close(mp4ff_t *ff);
-void mp4ff_track_add(mp4ff_t *f);
+static void mp4ff_track_add(mp4ff_t *f);
 int32_t parse_sub_atoms(mp4ff_t *f, const uint64_t total_size);
 int32_t parse_atoms(mp4ff_t *f);
 

diff -Naur faad2-orig/common/mp4v2/mp4property.h faad2/common/mp4v2/mp4property.h
--- faad2-orig/common/mp4v2/mp4property.h	2003-06-29 21:41:00.000000000 +0000
+++ faad2/common/mp4v2/mp4property.h	2005-08-29 15:32:44.000000000 +0000
@@ -21,6 +21,7 @@
 
 #ifndef __MP4_PROPERTY_INCLUDED__
 #define __MP4_PROPERTY_INCLUDED__
+#define NULL 0
 
 // forward declarations
 class MP4Atom;

diff -Naur faad2-orig/configure.in faad2/configure.in
--- faad2-orig/configure.in	2004-02-06 16:04:39.000000000 +0000
+++ faad2/configure.in	2005-08-29 15:32:04.000000000 +0000
@@ -67,7 +67,7 @@
         [HAVE_FPOS_T_POS])
 
 
-external_mp4v2=yes
+AC_CHECK_PROG(external_mp4v2, mpeg4ip-config, yes, no)
 AC_CHECK_LIB(mp4v2, MP4Create, , external_mp4v2=no, -lstdc++)
 AC_CHECK_LIB(mp4v2, MP4MetadataDelete, , external_mp4v2=no, -lstdc++)
 AC_CHECK_HEADER(mp4.h, , external_mp4v2=no)

diff -Naur faad2-orig/plugins/xmms/src/libmp4.c faad2/plugins/xmms/src/libmp4.c
--- faad2-orig/plugins/xmms/src/libmp4.c	2003-12-03 18:37:56.000000000 +0000
+++ faad2/plugins/xmms/src/libmp4.c	2005-08-29 15:35:24.000000000 +0000
@@ -34,6 +34,9 @@
 static int	mp4_isFile(char *);
 static void*	mp4Decode(void *);
 
+void getMP4info(char* file);
+int getAACTrack(MP4FileHandle file);
+
 InputPlugin mp4_ip =
   {
     0,	// handle
@@ -112,11 +115,13 @@
     gchar*	extention;
 
     extention = strrchr(filename, '.');
+
     if (extention &&
-	!strcasecmp(extention, ".mp4") ||	// official extention
-	!strcasecmp(extention, ".m4a") ||	// Apple mp4 extention
-	!strcasecmp(extention, ".aac")		// old MPEG2/4-AAC extention
-	){
+        (!strcasecmp(extention, ".mp4") ||	// official extention
+         !strcasecmp(extention, ".m4a") ||	// Apple mp4 extention
+         !strcasecmp(extention, ".aac")		  // old MPEG2/4-AAC extention
+        ))
+    {
       return (1);
     }
   }
